app/controllers/samples_controller.rb

```diff
@@ -150,7 +150,9 @@ def report_csv
   # GET /samples/1.json
 
   def show
+    @sample = samples_scope.find(params[:id])
```

Comment:
I don't think you need this. set_sample should be run by default.

Reply:
Resolved